primitives/runtime/src/runtime_string.rs

```diff
@@ -146,6 +146,13 @@ impl<'de> serde::Deserialize<'de> for RuntimeString {
 	}
 }
 
+#[cfg(all(not(feature = "std"), feature = "serde"))]
+impl<'de> serde::Deserialize<'de> for RuntimeString {
+	fn deserialize<D: serde::Deserializer<'de>>(de: D) -> Result<Self, D::Error> {
+		Vec::<u8>::deserialize(de).map(Self::Owned)
```

Comment:
```suggestion
		Ok(Self::Owned(serde::Deserialize::deserialize(de)?))
```

Then you should not require to duplicate the `Deserialize` implementation.

Reply:
Done in [3d4c295](https://github.com/paritytech/substrate/pull/13027/commits/3d4c29550e19db16b62133683bade8f387a5c7b9)